yellow-paper/docs/gas-and-fees/fee-payments-and-metering.md

```diff
@@ -0,0 +1,365 @@
+
+<!-- prettier-ignore -->
+| Value | Description |
+| -------- | -------- |
+| `feePerL1Gas` | Taken from the transaction |
+| `feePerL2Gas` | Taken from the transaction |
+| `feePerDAGas` | Taken from the transaction |
+| `totalFee` | Total fee provided, the product of all gas limits and fee per gas values specified in the transaction |
+| `l1GasUsed` | The accumulated quantity of L1 gas used, both amortized and per-transaction |
+| `l2GasUsed` | The accumulated quantity of L2 gas used, both amortized and per-transaction |
+| `DAGasUsed` | The accumulated quantity of DA gas used, both amortized and per-transaction |
+| `feeRecipient` | The aztec address designated as the recipient of fees for the current block |
+
+
+The values of gas used must be calculated and applied appropriately by the sequencer, a variety of constraints are in place for this.
+
+1. The sequencer specifies the size of rollup being produced to the base rollup circuit and uses this value when calculating amortized gas consumption. This value is a public input of the base rollup circuit.
+2. The sequencer specifies the fee recipient to the base rollup circuit and uses this value in fee distribution calls. This value is a public input of the base rollup circuit.
+3. The sequencer calculates an initial set of values for consumed transaction specific and amortized gas. 
+4. All forms of gas usage are accumulated by the public VM circuit and form part of the public inputs for the public kernel circuit.
+5. The public kernel circuit public inputs also include the gas and fee related inputs provided to the public VM circuit.
+6. The base rollup circuit computes the total amount of L1, L2 and DA gas consumed by the transaction, considering both private and public execution and transaction specific and amortized gas. It also considers reverted public execution. These values are public inputs to the circuit.
+7. The base rollup circuit verifies the fee distribution function was provided with correct values by constraining the appropriate public kernel public inputs.
+8. The values of rollup size, fee recipient and gas used are propagated up the rollup tree. The values of gas used are accumulated at each merge circuit and verified at the root rollup as not breaching the block gas limits. Rollup size and fee recipient are constrained in every merge circuit to be equal in the public inputs of the circuit's children.
+
+:::note
+TODO: Links to circuit pages and public inputs about the above
+:::
+
+## Block Limits
+
+The root rollup circuit will constrain block-wide limits on the quantities of gas that can be consumed. This is primarily to prevent state growth an ensure that network participants can sync to the head of the chain.
+
+## EIP-1559
+
+:::note
+This is a big TODO. We intend to implement some kind of 1559 system but it is still very much under discussion.
```

Comment:
Just adding that this might cause changes in the gas table above as we might have to introduce priority fees either per aggregate gas or per each gas type.

Reply:
Yes, it will.